middleware/middleware.js

```diff
@@ -38,36 +42,58 @@ module.exports.fetchRandomGameData = async (req, res, next) => {
         console.log(error.response.status);
       }
       if (retries === maxRetries) {
-        req.session.error = error.response.status;
-        res.redirect(ROUTES.error);
+        return error.response;
       }
     }
     retries++;
   }
-  if (response) {
-    req.gameData = response.data;
+  return response;
+};
+
+module.exports.fetchRandomGameData = async (req, res, next) => {
+  const maxPage = 6720;
 
-    next();
+  const response = await this.fetchFromApi(maxPage);
+  if (response.status >= 400) {
+    req.session.error = response.status;
+    res.redirect(ROUTES.error);
+  }
+  if (response) {
+    req.gameData = response.data.results[0];
   }
+
+  next();
 };
 
-// module.exports.fetchRandomGameDataArr = async (req, res, next) => {
-//   const pageNumArr = generateUniqueRandomArr();
-//   let gameDataArr = [];
-//   for (let num of pageNumArr) {
-//     let response = await axios.get(
-//       `${apiUrl}games?key=${process.env.RAWG_KEY}&page=${pageNumArr[num]}&${apiParams}`
-//     );
-//     response.data.results.forEach((result) => {
-//       if (result.background_image) {
-//         gameDataArr.push({ name: result.name, image: result.background_image });
-//       }
-//     });
-//   }
-//   console.log(gameDataArr);
-//   req.gameDataArr = gameDataArr;
-//   next();
-// };
+module.exports.fetchRandomGameDataArr = async (req, res) => {
+  const maxPage = 168;
+  const pageSize = 40;
+  const gameNumArr = generateUniqueRandomArr(40, 4);
+
+  let filteredGamesData = { gamesArray: [] };
+
+  const response = await this.fetchFromApi(maxPage, pageSize);
+
+  if (response.status >= 400) {
+    req.session.error = response.status;
+    res.redirect(ROUTES.error);
+  }
+
+  if (response) {
+    filteredGamesData.background_image =
+      response.data.results[gameNumArr[0]].background_image;
+    filteredGamesData.name = response.data.results[gameNumArr[0]].name;
+    gameNumArr.forEach((num) => {
+      const game = {
```

Comment:
There's not really much of a reason to declare a constant for every index, so I would recommend doing this instead:
```
filteredGamesData.gamesArray.push({
        gameName: response.data.results[num].name,
        gameId: response.data.results[num].id,
});
```